wcfsetup/install/files/lib/system/option/user/UserOptionHandler.class.php

```diff
@@ -224,7 +225,7 @@ protected function checkVisibility(Option $option) {
 		}
 		
 		// in registration
-		if ($this->inRegistration && !$option->askDuringRegistration && !$option->required && ($option->optionName != 'birthday' || !REGISTER_MIN_USER_AGE)) {
+		if ($this->inRegistration && !$option->askDuringRegistration && !$option->required && !($option->editable & UserOption::EDITABILITY_OWNER_ONLY_REGISTRATION) && ($option->optionName != 'birthday' || !REGISTER_MIN_USER_AGE)) {
```

Comment:
That wont work, because `wcf\data\user\option\UserOption` is not referenced in this class.